R/quantile_twas_weight.R

```diff
@@ -0,0 +1,485 @@
+#' @title Quantile TWAS Weight Calculation and QTL Analysis
+#'
+#' @description
+#' This file contains functions for performing Quantile Transcriptome-Wide 
+#' Association Studies (TWAS) weight calculations and Quantile QTL analysis.
+#' It provides tools for screening quantile regression results, performing
+#' LD clumping and pruning, and calculating TWAS weights.
+#'
+#' @details
+#' The main function in this file is `quantile_twas_weight_pipeline`, which
+#' orchestrates the entire analysis process. Other functions are helper
+#' functions used within the main pipeline.
+
+
+#' Qrank Score Test Screen
+#' @param X Matrix of predictors
+#' @param Y Matrix or vector of response variables
+#' @param Z Matrix of covariates (optional)
+#' @param tau.list Vector of quantiles to be analyzed
+#' @param threshold Significance threshold for adjusted p-values
+#' @param method Method for p-value adjustment ('fdr' or 'qvalue')
+#' @param top_count Number of top SNPs to select
+#' @param top_percent Percentage of top SNPs to select
+#' @return A list containing various results from the QR screen
+#' @export
+qr_screen <- function(X, Y, Z = NULL, tau.list, threshold = 0.05, method = 'qvalue', top_count = 10, top_percent = 15) {
+    p = ncol(X)
+    pvec = rep(NA, p)
+    ltau = length(tau.list)
+    quantile.pvalue = matrix(NA, nrow = p, ncol = ltau, dimnames = list(colnames(X), paste("p_qr", tau.list, sep = "_")))
+    quantile.zscore = matrix(NA, nrow = p, ncol = ltau, dimnames = list(colnames(X), paste("zscore_qr", tau.list, sep = "_")))
+    y = as.matrix(Y)
+    
+    if (!is.null(Z)) {
+        zz = cbind(rep(1, nrow(y)), Z)
+    } else {
+        zz = matrix(1, nrow = nrow(y), ncol = 1)
+    }
+
+    ranks_list = lapply(tau.list, function(tau) {
+        suppressWarnings(rq.fit.br(zz, y, tau = tau)$dual - (1 - tau))
```

Comment:
do you know why they suppressWarning? i worry by doing so we high something funky ..